crates/solidity/testing/sanctuary/src/main.rs

```diff
@@ -60,10 +60,10 @@ fn process_dataset(dataset: &impl Dataset, versions: &BTreeSet<Version>) -> Resu
 
     let total_errors = reporter.finish();
     if total_errors > 0 {
-        std::process::exit(1);
+        bail!("There were errors processing the dataset.")
```

Comment:
Returning an error from `main` already quits with error code 1, no need to do that ourselves explicitly.

Reply:
This is different, as returning an error from main (for an internal error/crash/bug) will print a ton of stack traces, making it hard to actually see the syntax errors (output of the tool). If we got to this point, we just want to print the syntax errors and exit without panics/traces (hence the `exit(1)`).

Maybe I should add a comment here about this.

Reply:
This is different, as returning an error from main (for an internal error/crash/bug) will print a ton of stack traces, making it hard to actually see the syntax errors (output of the tool). If we got to this point, we just want to print the syntax errors and exit cleanly without panics/traces (hence the `exit(1)`).

Maybe we should add a comment here about this.